gdshelpers/parts/port.py

```diff
@@ -113,17 +113,26 @@ def angle(self, angle):
     @property
     def width(self):
         """
-        The width of the port.
+        The width of the port. E.g. for slot waveguides it can also be an array in the format [width, gap, width, ...],
+        where each width describe the width of each rail and the gap defines the gap in between.
+        This array can also end with a gap, which facilitates e.g. the design of adiabatic mode converters.
+        """
+        return self._width
+
+    @property
+    def total_width(self):
+        """
+        The total width of the port.
 
         Guarantied to be a positive float.
         """
-        return self._width
+        return np.sum(self._width)
 
     # noinspection PyAttributeOutsideInit
     @width.setter
     def width(self, width):
-        assert width > 0, 'Port width must be larger than zero'
-        self._width = float(width)
+        assert np.sum(width) > 0, 'Port width must be larger than zero'
+        self._width = np.array(width)
```

Comment:
This means that the width is *always* an array, even for singular values...

Reply:
okay, now it checks if width has more than one element and only saves it in this case as an array